include/Messenger.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Messenger: class to abstract the routing of Open Sound Control messages from the core
+ *   to another area of the core and/or the gui
+ *
+ * (c) 2015 Colin Wallace (https://github.com/Wallacoloo)
+ *
+ * This file is part of LMMS - http://lmms.io
+ *
+ * This file is released under the MIT License.
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef MESSENGER_H
+#define MESSENGER_H
+
+#include <cstring>
+
+#include <QReadWriteLock>
+#include <QSet>
+
+class QString;
+
+class OscMsgListener;
+
+
+class Messenger
+{
+public:
+	// const strings to specify the Open Sound Control endpoint locations,
+	// e.g. "/mixer/ch/n/volume"
+	class Endpoints
+	{
+	public:
+		// Endpoints to send general and already-translated warnings / error messages
+		static const char *Warning;
+		static const char *Error;
+
+		static const char *WaveTableInit;
+		static const char *MixerDevInit;
+		static const char *MixerProcessingStart;
+	};
+
+	// Send message to indicate that the following components have completed their initialization
+	void broadcastWaveTableInit();
```

Comment:
@Wallacoloo I definitively agree on keeping all the string constants in one file and not have them scattered as hard coded strings in several files. If possible these strings should be implementation details that are not visible to the outside.

Concerning my remarks about methods like `broadcastWaveTableInit`: I was wondering why any component should need to know that the wavetables have finished being initialized. I would assume that things like (global) wavetables should be initialized before the components are wired together in the first place. Or am I missing some important detail?